definitions/checks/system_registration.rb

```diff
@@ -0,0 +1,27 @@
+class Checks::SystemRegistration < ForemanMaintain::Check
+  metadata do
+    label :verify_self_registration
+    description 'Check whether system is self-registered or not'
+    tags :default
+
+    confine do
+      file_exists?('/etc/rhsm/rhsm.conf')
+    end
+  end
+
+  def run
+    if system_is_self_registerd?
+      raise ForemanMaintain::Error::Warn, 'System is self registered'
+    else
+      puts 'System is not self registered'
+    end
+  end
+
+  def system_is_self_registerd?
+    rhsm_hostname.casecmp(hostname).zero?
+  end
+
+  def rhsm_hostname
+    execute("grep '\\bhostname\\b' < /etc/rhsm/rhsm.conf").sub(/.*?=/, '').strip
```

Comment:
While testing, I hit sill one issue, when I had a line commented out with this value. `'^\\s*\\bhostname\\b'` to match online uncommented line.